integration_tests/src/main/python/asserts.py

```diff
@@ -312,52 +312,70 @@ def assert_gpu_fallback_collect(func,
 
 def _assert_gpu_and_cpu_are_equal(func,
         mode,
+        is_cpu_first=True,
```

Comment:
New arguments should not appear before other default args, or we risk existing code passing a conf as the third argument and it now being interpreted as the `is_cpu_first` argument.  Code can still call it with `is_cpu_first=False` if they don't want to pass a conf explicitly.